chain/store/store.go

```diff
@@ -170,7 +172,9 @@ func (cs *ChainStore) SubHeadChanges(ctx context.Context) chan []*HeadChange {
 				case <-ctx.Done():
 				}
 			case <-ctx.Done():
```

Comment:
this will trigger every loop, but since the channel above should be closed immediately it's probably not a huge issue

Reply:
it actually caused a huge memory increasing when i had 100 clients closed once, like 10x.
i think it's because the undelying chan sending
https://github.com/whyrusleeping/pubsub/blob/master/pubsub.go#L78-L85

Reply:
@magik6k